src/renderer/viz/expressions/RampImage.js

```diff
@@ -0,0 +1,88 @@
+import defaultSVGs from '../defaultSVGs';
+import { checkType, checkInstance } from './utils';
+import Property from './basic/property';
+import CategoryIndex from './CategoryIndex';
+import ListImage from './ListImage';
+import SVG from './SVG';
+import Base from './base';
+import { OTHERS_GLSL_VALUE, DEFAULT_OPTIONS, DEFAULT_OTHERS } from './constants';
+
+export default class RampImage extends Base {
+    _bindMetadata (metadata) {
+        Base.prototype._bindMetadata.call(this, metadata);
+        this.type = this.palette.childType;
+
+        if (this.others === DEFAULT_OTHERS) {
+            this.others = new SVG(defaultSVGs.circle);
+        } else {
+            checkType('ramp', 'others', 2, 'image', this.others);
+        }
+
+        if (this.input.isA(Property)) {
+            this.input = new CategoryIndex(this.input);
+            this.input._bindMetadata(metadata);
+        }
+
+        checkType('ramp', 'input', 0, 'category', this.input);
+        checkInstance('ramp', 'palette', 1, ListImage, this.palette);
+
+        this.others._bindMetadata(metadata);
+        this.childrenNames.push('others');
+    }
+
+    eval (feature) {
+        const index = this.input.eval(feature);
+        return this._calcEval(index);
+    }
+
+    _calcEval (input) {
+        const index = Math.round(input * (this.input.numCategoriesWithoutOthers - 1));
+        const paletteValues = this.palette.eval();
+        return paletteValues[index] ? paletteValues[index] : this.others.eval();
+    }
+
+    getLegend (options) {
+        const config = Object.assign({}, DEFAULT_OPTIONS, options);
+        const type = this.input.type;
+        const legendData = this.input.getLegendData(config);
+        const data = legendData.data.map(({key, value}) => {
+            value = this._calcEval(value);
+            return { key, value };
+        });
+
+        return { type, ...legendData, data };
+    }
+
+    _applyToShaderSource (getGLSLforProperty) {
+        const input = this.input._applyToShaderSource(getGLSLforProperty);
+        const images = this.palette._applyToShaderSource(getGLSLforProperty);
+        const others = this.others._applyToShaderSource(getGLSLforProperty);
+        return {
+            preface: this._prefaceCode(`
+                uniform float rampImageMultiplier${this._uid};
+
+                ${input.preface}
+                ${images.preface}
+                ${others.preface}
+
+                vec4 rampImage${this._uid}(float index, vec2 imageUV){
```

Comment:
If imagelist function prototype is `float index, vec2 uv` and rampImage function is `vec2 uv, float index` we should change one of them to be consistent